src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,320 @@
+    uint8_t mode = (BMP388_MODE_FORCED << 4) | (1 << 1) | (1 << 0);
+    busWriteRegister(busdev, BMP388_PWR_CTRL_REG, mode);
+}
+
+bool bmp388Detect(const bmp388Config_t *config, baroDev_t *baro)
+{
+    delay(20);
+
+#if defined(BARO_EXTI_PIN)
+    IO_t baroIntIO = IO_NONE;
+#endif
+
+#if defined(BARO_EXTI_PIN) && defined(USE_EXTI)
+    if (config && config->baroInitExtiTag) {
+        baroIntIO = IOGetByTag(config->baroInitExtiTag);
+        IOInit(baroIntIO, OWNER_BARO_EXTI, 0);
+        EXTIHandlerInit(&baro->exti, bmp388_extiHandler);
+        EXTIConfig(baroIntIO, &baro->exti, NVIC_PRIO_BARO_EXTI, IOCFG_IN_FLOATING, EXTI_TRIGGER_RISING);
+        EXTIEnable(baroIntIO, true);
+    }
+#else
+    UNUSED(config);
+#endif
+
+    busDevice_t *busdev = &baro->busdev;
+    bool defaultAddressApplied = false;
+
+    bmp388BusInit(busdev);
+
+    if ((busdev->bustype == BUSTYPE_I2C) && (busdev->busdev_u.i2c.address == 0)) {
+        // Default address for BMP388
+        busdev->busdev_u.i2c.address = BMP388_I2C_ADDR;
+        defaultAddressApplied = true;
+    }
+
+    busReadRegisterBuffer(busdev, BMP388_CHIP_ID_REG, &bmp388_chip_id, 1);
+
+    if (bmp388_chip_id != BMP388_DEFAULT_CHIP_ID) {
+        bmp388BusDeinit(busdev);
+        if (defaultAddressApplied) {
+            busdev->busdev_u.i2c.address = 0;
+        }
+        return false;
+    }
+
+    // read calibration
+    STATIC_ASSERT(sizeof(bmp388_calib_param_t) == BMP388_TRIMMING_DATA_LENGTH, bmp388_calibration_structure_incorrectly_packed);
+
+#if defined(BARO_EXTI_PIN) && defined(USE_EXTI)
+    uint8_t intCtrlValue = (1 << BMP388_INT_DRDY_EN_BIT) | (0 << BMP388_INT_FFULL_EN_BIT) | (0 << BMP388_INT_FWTM_EN_BIT) | (0 << BMP388_INT_LATCH_BIT) | (1 << BMP388_INT_LEVEL_BIT) | (0 << BMP388_INT_OD_BIT);
+    busWriteRegister(busdev, BMP388_INT_CTRL_REG, intCtrlValue);
+#endif
+
+    busReadRegisterBuffer(busdev, BMP388_TRIMMING_NVM_PAR_T1_LSB_REG, (uint8_t *)&bmp388_cal, sizeof(bmp388_calib_param_t));
+
+
+    // set oversampling
+    busWriteRegister(busdev, BMP388_OSR_REG,
+        ((BMP388_PRESSURE_OSR << BMP388_OSR_P_BIT) & BMP388_OSR_P_MASK) |
+        ((BMP388_TEMPERATURE_OSR << BMP388_OSR4_T_BIT) & BMP388_OSR4_T_MASK)
+    );
+
+    bmp388BeginForcedMeasurement(busdev);
+
+    // these are dummy as temperature is measured as part of pressure
+    baro->ut_delay = 0;
+    baro->get_ut = bmp388_get_ut;
+    baro->start_ut = bmp388_start_ut;
+    // only _up part is executed, and gets both temperature and pressure
+    baro->start_up = bmp388_start_up;
+    baro->get_up = bmp388_get_up;
+
+    // See datasheet 3.9.2 "Measurement rate in forced mode and normal mode"
+    baro->up_delay = 234 +
+        (392 + (powerf(2, BMP388_PRESSURE_OSR + 1) * 2000)) +
+        (313 + (powerf(2, BMP388_TEMPERATURE_OSR + 1) * 2000));
+
+    baro->calculate = bmp388_calculate;
+
+    return true;
+}
+
+static void bmp388_start_ut(baroDev_t *baro)
```

Comment:
Function names should be camelCase. See https://github.com/betaflight/betaflight/blob/master/docs/development/CodingStyle.md

Reply:
I'm following the pattern defined by the other baro drivers.  suggest a separate PR to update them all at the same time.